### ISSUE TITLE
Expose a Design Document

### ISSUE THREAD

**Issue by mykeels** [changeweb/Unifiedtransform#119]
I'd like to contribute when I can, but it's hard knowing your intentions for parts of the application that haven't been written yet.

Could you provide a document that gives the design of the application? A good and simple format to begin with is:

## Class

This is a description of a school's class. It does this and that ... Its properties are:

- Property 1
- Property 2

## Section

A class can have multiple sections. It represents the divisions within a single class ... It's properties are:

- Property 1
- Property 2

**Comment by changeweb** [comment]
@mykeels you are right. A design document is necessary. Should I write the document in Wiki or Issue? Which one will be convenient for all?

Thanks for your enthusiasm.

**Comment by mykeels** [comment]
A Wiki is fine. That's where we go to, when we can't find what we're looking for in the README.md

**Comment by changeweb** [comment]
I also want some parts to be modular so that we can support different schools' situation e.g.: Different grading and marking system.

What do you think?

**Comment by mykeels** [comment]
Cool!

Right now, I'm thinking those can be functional, a clearly defined function that can be overridden by adopters to provide custom functionality.

But I don't know your idea for the grading system yet, so this might not fit in. If you explained, I might be able to help. How does the grading system work currently?

**Comment by changeweb** [comment]
@mykeels I'm sorry for not being able to do the works because unfortunately my windows partition that I've been working in, have become unfixable. :sad:

I will try to provide info using my phone.

**Comment by changeweb** [comment]
I've added some pages in wiki. You may give your feedback on these. It will help me to improve further documentation.

**Comment by mykeels** [comment]
The layout looks okay. 

- The bullet points in the "Design Document" page should link to their respective pages.
- Are "Notice, Event, Syllabus, Routine" together as one module? If not, shouldn't they be separate?

**Comment by changeweb** [comment]
@mykeels Exam document page added. I'm trying to continue documentation with software improvement. So it is getting bit slow :/

* The bullet points links are being added with each page creation.
* Right now "Notice, Event, Syllabus, Routine" are considered as one module. But in near future they will have their own special features.

Thanks for your feedback.

**Comment by safetoleave** [comment]
Is there a feature where you can attach a fingerprint scanner to record the teacher's attendance?
